sympy/core/relational.py

```diff
@@ -434,7 +434,10 @@ def __new__(cls, lhs, rhs=None, **options):
                 if L is False:
                     if lhs == -rhs:  # Eq(oo, -oo)
                         return S.false
-                    return S.true
+                    L, R = [i.is_extended_positive for i in (lhs, rhs)]
+                    if L == R and L != None:
+                        return S.true
```

Comment:
One consequence of this return (same as master) is:
```julia
In [4]: Eq(zoo, I*oo)                                                                                                             
Out[4]: True
```
Another:
```julia
In [6]: x = 1+I*oo                                                                                                                

In [7]: y = 2+I*oo                                                                                                                

In [8]: x                                                                                                                         
Out[8]: 1 + ∞⋅ⅈ

In [9]: y                                                                                                                         
Out[9]: 2 + ∞⋅ⅈ

In [10]: Eq(x, y)                                                                                                                 
Out[10]: True

In [11]: x - y                                                                                                                    
Out[11]: nan

In [12]: re(x) == re(y)                                                                                                           
Out[12]: False
```
Perhaps this should check extended_real before extended_positive...